packages/fxa-settings/src/pages/InlineTotpSetup/index.tsx

```diff
@@ -0,0 +1,320 @@
+                          <button
+                            onClick={() => {
+                              setShowQR(true);
+                            }}
+                            className="link-blue inline"
+                          >
+                            Scan QR code instead?
+                          </button>
+                        ),
+                      }}
+                    >
+                      <p className="text-sm m-2">
+                        Type this secret key into your authentication app.{' '}
+                        <button
+                          onClick={() => {
+                            setShowQR(true);
+                          }}
+                          className="link-blue inline"
+                        >
+                          Scan QR code instead?
+                        </button>
+                      </p>
+                    </FtlMsg>
+                    <div className="qr-code-container">
+                      <div className="qr-code-text">{secret}</div>
+                    </div>
+                    <FtlMsg id="inline-totp-setup-on-completion-description">
+                      <p className="text-sm m-4">
+                        Once complete, it will begin generating security codes
+                        for you to enter.
+                      </p>
+                    </FtlMsg>
+                  </>
+                )}
+                <form
+                  noValidate
+                  className="flex flex-col gap-4"
+                  onSubmit={handleSubmit(onSubmit)}
+                >
+                  {/* Using `type="text" inputmode="numeric"` shows the numeric pad on mobile and strips out whitespace on desktop. */}
+                  {/* TODO: standardize with input used for Settings/TwoStepAuthentication? */}
+                  <InputText
+                    type="text"
+                    inputMode="numeric"
+                    label={localizedSecurityCodePlaceholder}
+                    onChange={(e) => {
+                      setTotpCodeValue(e.target.value);
+                      // clear error tooltip if user types in the field
+                      if (totpErrorMessage) {
+                        setTotpErrorMessage('');
+                      }
+                    }}
+                    onFocusCb={onFocusMetricsEvent ? onFocus : undefined}
+                    errorText={totpErrorMessage}
+                    autoFocus
+                    pattern="\d{6}"
+                    className="text-start"
+                    anchorStart
+                    autoComplete="off"
+                    spellCheck={false}
+                    prefixDataTestId="inline-totp-setup-code"
+                    required
+                    tooltipPosition="bottom"
+                    maxLength={6}
+                  />
+                  <button
+                    type="submit"
+                    className="cta-primary cta-xl w-full my-2"
```

Comment:
<img width="586" alt="image" src="https://user-images.githubusercontent.com/13018240/216117460-a0598826-f778-4c4a-a5c2-43e93d9ec926.png">

You shouldn't need `w-full` here given the `cta-xl` class.

Also this is a nit but can we put a `margin-bottom` on this instead of `my-2` to match other button styles or put it on the "cancel setup" link? Also don't have a screenshot handy of the old view but I feel like we might could use more space between "cancel setup" and the input? Maybe we can reference other views for spacing.

Reply:
Hmm for whatever reason it does need the `w-full` class -- I've seen this in a few other PRs also. I'll change the margin bottom!

Reply:
Yessss this got removed by using the `FormVerifyCode` element!